src/analyses/memOutOfBounds.ml

```diff
@@ -0,0 +1,344 @@
+open GoblintCil
+open Analyses
+open MessageCategory
+
+module AS = AnalysisState
+module VDQ = ValueDomainQueries
+
+module Spec =
```

Comment:
Maybe add a comment that this is an analysis only to be able to have a separation of concerns and to be able to conveniently turn it on or off, but that it doesn't track any internal state.

Reply:
Added an explanation comment in https://github.com/goblint/analyzer/pull/1094/commits/924130132a766149cc97745629fd52e08ec6b798